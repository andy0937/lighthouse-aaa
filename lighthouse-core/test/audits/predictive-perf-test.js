/**
 * @license Copyright 2017 Google Inc. All Rights Reserved.
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
 */
'use strict';

const PredictivePerf = require('../../audits/predictive-perf.js');

const acceptableTrace = require('../fixtures/traces/progressive-app-m60.json');
const acceptableDevToolsLog = require('../fixtures/traces/progressive-app-m60.devtools.log.json');

/* eslint-env jest */
describe('Performance: predictive performance audit', () => {
  it('should compute the predicted values', () => {
    const artifacts = {
      traces: {
        [PredictivePerf.DEFAULT_PASS]: acceptableTrace,
      },
      devtoolsLogs: {
        [PredictivePerf.DEFAULT_PASS]: acceptableDevToolsLog,
      },
    };

    return PredictivePerf.audit(artifacts, {computedCache: new Map()}).then(output => {
      const metrics = output.details.items[0];
      for (const [key, value] of Object.entries(metrics)) {
        metrics[key] = Math.round(value);
      }

      expect(metrics).toMatchSnapshot();
    });
  });
});
