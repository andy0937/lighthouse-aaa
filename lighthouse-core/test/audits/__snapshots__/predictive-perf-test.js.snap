// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Performance: predictive performance audit should compute the predicted values 1`] = `
Object {
  "optimisticEIL": 93,
  "optimisticFCP": 1307,
  "optimisticFMP": 1439,
  "optimisticSI": 605,
  "optimisticTTFCPUI": 3351,
  "optimisticTTI": 3351,
  "pessimisticEIL": 101,
  "pessimisticFCP": 1307,
  "pessimisticFMP": 1583,
  "pessimisticSI": 1631,
  "pessimisticTTFCPUI": 3502,
  "pessimisticTTI": 3502,
  "roughEstimateOfEIL": 78,
  "roughEstimateOfFCP": 1307,
  "roughEstimateOfFMP": 1511,
  "roughEstimateOfSI": 1657,
  "roughEstimateOfTTFCPUI": 3351,
  "roughEstimateOfTTI": 3427,
}
`;
